src/test/java/io/confluent/connect/jdbc/JdbcSinkConnectorTest.java

```diff
@@ -0,0 +1,70 @@
+package io.confluent.connect.jdbc;
+
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.config.Config;
+
+import static io.confluent.connect.jdbc.sink.JdbcSinkConfig.PK_MODE;
+import static org.junit.Assert.*;
+
+import org.apache.kafka.common.config.ConfigValue;
+import org.junit.Test;
+
+public class JdbcSinkConnectorTest {
+
+  @Test
+  public void testValidationWhenDeleteEnabled() {
+
+    Map<String, String> connConfig = new HashMap<>();
+    connConfig.put("connector.class", "io.confluent.connect.jdbc.JdbcSinkConnector");
+    connConfig.put("delete.enabled", "true");
+    connConfig.put("pk.mode", "none");
+
+    JdbcSinkConnector connector = new JdbcSinkConnector();
+
+    List<String> pkModeError = connector.validate(connConfig)
+        .configValues()
+        .stream()
+        .filter(cfg -> PK_MODE.equals(cfg.name()))
+        .map(ConfigValue::errorMessages)
+        .findFirst()
+        .orElse(Collections.emptyList());
+
+    assertFalse(pkModeError.isEmpty());
+
+    connConfig.put("pk.mode", "record_key");
+
+    pkModeError = connector.validate(connConfig)
+        .configValues()
+        .stream()
+        .filter(cfg -> PK_MODE.equals(cfg.name()))
+        .map(ConfigValue::errorMessages)
+        .findFirst()
+        .orElse(Collections.emptyList());
+
+    assertTrue(pkModeError.isEmpty());
+  }
+
+  @Test
+  public void testValidationWhenDeleteNotEnabled() {
+
+    Map<String, String> connConfig = new HashMap<>();
+    connConfig.put("connector.class", "io.confluent.connect.jdbc.JdbcSinkConnector");
+    connConfig.put("pk.mode", "none");
```

Comment:
Nit: may want to explicitly disable deletes here (makes the test a little easier to read for people who don't know the default off the top of their head, and makes it resilient to potential changes to that default)

```suggestion
    connConfig.put("pk.mode", "none");
    connConfig.put("delete.enabled", "false");
```

Reply:
good point.